core/server/master/src/main/java/alluxio/master/FaultTolerantAlluxioMasterProcess.java

```diff
@@ -73,16 +73,25 @@ protected FaultTolerantAlluxioMasterProcess(JournalSystem journalSystem,
   public void start() throws Exception {
     mRunning = true;
     mJournalSystem.start();
+
+    startMasters(false);
+    LOG.info("Secondary started");
+
+    if (ServerConfiguration.getBoolean(PropertyKey.MASTER_JOURNAL_CATCHUP_PROTECT_ENABLED)) {
+      mJournalSystem.waitForCatchup();
```

Comment:
Do we still need it here now that we do it anyway within the while loop?

Reply:
Yeah it's still needed. We want the initial replay (which could be really long in production env) to finish before joining the leader election.
If this is removed, there could be a case that
1. Master A is the previous leading master
2. Master B joins the leader election
3. Master B starts the journal initial replay, during the initial replay, Master B is selected as the primary master
4. All the clients need to wait for master B to finish its initial replay and start serving requests even though other masters are available.

In the beginning, there will be two `wait for catch up` in a row, the second one is usually a noop and finish in one second. This doesn't seem to introduce any issues

Reply:
That makes sense. Let's keep both then.

Reply:
Sounds good.

Can we add a short comment here describing that? Maybe something like `// Perform initial catchup before joining leader election, to avoid potential delay if this process is selected as leader`